src/main/java/org/zalando/nakadi/service/EventPublisher.java

```diff
@@ -82,6 +95,17 @@ public EventPublishResult publish(final String events, final String eventTypeNam
         } catch (final EventPublishingException e) {
             LOG.error("error publishing event", e);
             return failed(batch);
+        } catch (InterruptedException | TimeoutException e) {
```

Comment:
After catching InterruptedException one must do one of the following:
 - stop the thread
 - rise InterruptedException again
 - Call Thread.interrupt(), so the thread owner will be able to detect thread interruption